.github/workflows/integration.yml

```diff
@@ -1,61 +1,117 @@
+    types:
+      - labeled
+      - opened
+      - reopened
+      - synchronize
+      - unlabeled
 
 jobs:
+  splitter:
+    runs-on: ubuntu-latest
+    env:
+      source_dir: "cloud_awsops"
+    outputs:
+      test_targets: ${{ steps.display.outputs.test_targets }}
+    steps:
+      - name: Checkout collection
+        uses: actions/checkout@v3
+        with:
+          path: ${{ env.source_dir }}
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: "0"
+
+      - name: List changes for pull request
+        id: splitter
+        uses: ansible-network/github_actions/.github/actions/ansible_test_splitter@main
+        with:
+          collections_to_test: ${{ env.source_dir }}
+          total_jobs: 4
+
+      - name: display targets
+        id: display
+        run: echo "test_targets=${{ steps.splitter.outputs.test_targets }}" >> $GITHUB_OUTPUT
+        shell: bash
+  build_matrix:
+    runs-on: ubuntu-latest
+    if: ${{ needs.splitter.outputs.test_targets != '' }}
+    needs:
+      - splitter
+    outputs:
+      test_matrix: ${{ steps.generate-workflow.outputs.targets }}
+      test_targets: ${{ needs.splitter.outputs.test_targets }}
+    steps:
+      - name: Generate workflow id
+        id: generate-workflow
+        run: |
+          import os, json
+          targets=[d.split(':', maxsplit=1)[0] for d in os.environ.get('TEST_TARGETS').split(";")]
+          output_path = os.environ.get('GITHUB_OUTPUT')
+          with open(output_path, "a", encoding="utf-8") as fw:
+              fw.write(f'targets={json.dumps(targets)}\n')
+        shell: python
+        env:
+          TEST_TARGETS: ${{ needs.splitter.outputs.test_targets }}
+
+      - name: Display matrix elements
+        run: echo "Workflow list => ${{ steps.generate-workflow.outputs.targets }}"
+        shell: bash
   test:
     runs-on: ubuntu-latest
+    needs:
+      - build_matrix
     env:
       source: "./source"
-      aws_dir: "./amazon_aws"
-      crypto_dir: "./community.crypto"
       ansible_version: "stable-2.14"
       python_version: "3.9"
+      SPLITTER_TARGETS: ${{ needs.build_matrix.outputs.test_targets }}
+    strategy:
+      fail-fast: false
+      matrix:
+        workflow-id: ${{ fromJson(needs.build_matrix.outputs.test_matrix) }}
+    name: "integration-${{ matrix.workflow-id }}"
     steps:
+      - name: Read ansible-test targets
+        id: read-targets
+        run: |
+          import os
+          workflow_id = os.environ.get('TEST_WORKFLOW_ID')
+          all_targets = os.environ.get('SPLITTER_TARGETS')
+          workflow_targets={d.split(':', maxsplit=1)[0]: d.split(':', maxsplit=1)[1] for d in all_targets.split(";")}[workflow_id]
+          targets = " ".join(workflow_targets.split(","))
+          output_path = os.environ.get('GITHUB_OUTPUT')
+          with open(output_path, "a", encoding="utf-8") as fw:
+              fw.write(f"ansible_test_targets={targets}\n")
+        shell: python
```

Comment:
This and the entire build_matrix job seems like something the ansible_test_splitter action should be doing. The test splitter's job should be to output something that's usable in GHA, not something that then has to be processed.

Reply:
@gravesm 
fix: ansible_test_splitter has been updated via https://github.com/ansible-network/github_actions/pull/115 to produce simplified output for workflow
Please review this PR before this one can be merged

Reply:
updated